src/js/panels.js

```diff
@@ -198,16 +193,317 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+                args: ["{that}.bindPrev"]
+            },
+            "afterRender.bindNext": {
+                "this": "{that}.dom.next",
+                method: "click",
+                args: ["{that}.bindNext"]
+            },
+            "afterRender.setButtonStates": {
+                funcName: "gpii.firstDiscovery.panel.lang.setNavKeyStatus",
+                args: ["{that}"]
+            },
+            "afterRender.getButtonTops": {
+                funcName: "gpii.firstDiscovery.panel.lang.getButtonTops",
+                args: ["{that}"]
+            },
+            // To override the default scrolling behavior from buttons' parent overflow div to make sure when using keyboard to focus
+            // on the button, the overflow div scrolls to the calculated position.
+            "afterRender.overrideDefaultScroll": {
+                funcName: "gpii.firstDiscovery.panel.lang.overrideDefaultScroll",
+                args: ["{that}"]
+            },
+            "afterRender.scrollLangIntoView": {
+                funcName: "gpii.firstDiscovery.panel.lang.scrollLangIntoView",
+                args: ["{that}"]
+            },
+            "onButtonTopsReady.scrollLangIntoView": {
+                funcName: "gpii.firstDiscovery.panel.lang.scrollLangIntoView",
+                args: ["{that}"]
+            },
+            "afterRender.preventWrapWithArrowKeys": {
+                funcName: "gpii.firstDiscovery.panel.lang.preventWrapWithArrowKeys",
+                args: ["{that}"]
+            }
+        }
+    });
+
+    // This component is needed for the following demands block to be only applied to the language panel "gpii.firstDiscovery.panel.lang".
+    // Without this component being the sub-component of the language panel, according to http://wiki.fluidproject.org/display/docs/Contexts,
+    // when the context component of the demands block was the language panel itself, the demands block would also be applied to siblings of
+    // the language panel. To work around this issue, another layer of containment needs to be added.
+    // This component and the demands block should be removed when the new framework (http://issues.fluidproject.org/browse/FLUID-5249)
+    // is in use.
+    fluid.defaults("gpii.firstDiscovery.panel.lang.attachTooltipOnLang", {
+        gradeNames: ["gpii.firstDiscovery.attachTooltip", "autoInit"],
+        tooltipContentMap: {
+            "prev": "navButtonTooltip",
+            "next": "navButtonTooltip"
+        }
+    });
+
+    fluid.demands("fluid.tooltip", ["gpii.firstDiscovery.panel.lang.attachTooltipOnLang"], {
+        options: {
+            styles: {
+                tooltip: "gpii-fd-tooltip-lang"
+            }
+        }
+    });
+
+    gpii.firstDiscovery.panel.lang.setNavKeyStatus = function (that) {
+        var langArray = that.options.controlValues.lang,
+            selectedLang = that.model.lang;
+
+        that.locate("prev").prop("disabled", selectedLang === langArray[0]);
```

Comment:
This should be modelised into two flags and updated via model listeners to lang (it could even be done via relay using the fluid.transforms.binaryOp with operator of "==="